Automattic/jetpack
# Google Fonts: Integrate with new font library

Related to https://github.com/Automattic/wp-calypso/issues/81875

## Proposed changes:

* As suggested, we served [the collection of the Jetpack Google Fonts on wpcom](https://s0.wp.com/i/font-collections/jetpack-google-fonts.json). The collection is generated via the `generate_google_fonts` CLI tool on wpcom.
* Next, this PR gets the collection data from the remote and injects the fonts to the default global styles settings on the fly by the `wp_theme_json_data_default` hook so we can display Google Fonts in the dropdown of the Font Family.
* Why the default?
  * The theme global styles will be overridden after people pick any style variation. It leads to the font family from the theme global styles disappearing. Hence, it would be better to use the default global styles instead of the theme. See https://github.com/Automattic/dotcom-forge/issues/3162, https://github.com/Automattic/wp-calypso/issues/78605 for more context.
  * There is a [WIP PR](https://github.com/WordPress/gutenberg/pull/54888) to allow developers to register the default fonts and display them in the “Default Fonts” section. The benefits are the default fonts are activated by default and cannot be configurable. So, when this module is deactivated, the fonts provided by this module won't stick in the user's global styles settings. ~However, the default fonts are invisible on the block settings as [the block settings pick font families from one of the origins](https://github.com/WordPress/gutenberg/issues/55011#issuecomment-1748198200). But I assume it's acceptable for now since the user will only see the installed fonts there if they install a font. As a result, I think it makes to inject them into the default now~ (The issue is resolved 🎉)

![image](https://github.com/Automattic/jetpack/assets/13596067/990693da-c7c2-4600-a9a4-a490b01aca8b)

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Create a [Jurassic Ninja sandbox](https://jurassic.ninja/create/?jetpack-beta&branch=feat/google-fonts-with-new-font-library) against this PR.
* Install [the latest Gutenberg](https://github.com/WordPress/gutenberg/releases), e.g.: v16.7.0-rc.1
* Go to `/wp-admin/admin.php?page=jetpack_modules` to activate the “Google Fonts (Beta)” module
* Go to the Site Editor
* Select Styles > Typography > Text > Font
* Ensure you're able to see all Google Fonts in the list
* Disable the Gutenberg plugin
* Select Styles > Typography > Text > Font again
* Ensure you can still see all Google Fonts in the list
